### PR TITLE
Throw error when fetching an object without ID

### PR DESCRIPTION
**Pull request by percypyan**
Pull Request fixing issue #1063.

Add a check for ID definition before fetching data.

### CODE REVIEW

**Comment by percypyan** [comment]
@dplewis Maybe should I only throw for "" ID to avoid breaking code fetching with `null` or `undefined` ID?

File: src/ParseObject.js [hunk 1]
```diff
@@ -179,10 +179,10 @@ class ParseObject {
    * Returns a local or server Id used uniquely identify this object
    */
   _getId(): string {
-    if (typeof this.id === 'string') {
+    if (this.id && typeof this.id === 'string') {
```
Comment (dplewis) [inline src/ParseObject.js]: Can you revert this change? I don't think this is needed. You can write integration test to verify.

Comment (percypyan) [reply src/ParseObject.js]: @dplewis as you wish I've written a unit test but it's maybe useless to add this check

Comment (dplewis) [reply src/ParseObject.js]: An integration tests is different from unit tests. Integration tests run against a parse server instance. Add your tests [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/integration/test/ParseObjectTest.js).
